transfers/amclient.py

```diff
@@ -229,21 +49,30 @@ def _call_url_json(url, params, method='GET'):
     """
     method = method.upper()
     LOGGER.debug('URL: %s; params: %s; method: %s', url, params, method)
-    response = requests.request(method, url=url, params=params)
-    LOGGER.debug('Response: %s', response)
-    LOGGER.debug('type(response.text): %s ', type(response.text))
-    LOGGER.debug('Response content-type: %s', response.headers['content-type'])
-    if not response.ok:
-        LOGGER.warning('%s Request to %s returned %s %s', method, url,
-                       response.status_code, response.reason)
-        LOGGER.debug('Response: %s', response.text)
-        return None
+
     try:
-        return response.json()
-    except ValueError:  # JSON could not be decoded
-        LOGGER.warning('Could not parse JSON from response: %s',
-                       response.text)
-        return None
+        response = requests.request(method, url=url, params=params)
+        LOGGER.debug('Response: %s', response)
+        LOGGER.debug('type(response.text): %s ', type(response.text))
+        LOGGER.debug('Response content-type: %s',
+                     response.headers['content-type'])
+
+        if not response.ok:
+            LOGGER.warning('%s Request to %s returned %s %s', method, url,
+                           response.status_code, response.reason)
+            LOGGER.debug('Response: %s', response.text)
+            return errors.ERR_INVALID_RESPONSE
```

Comment:
Good spot on the docstring. 

I like the `utils.py` idea as well. I have to change the error_lookup to be a little more robust - a function call rather than a naive dictionary lookup but I think that will work well across both scripts.